src/cart/crt_init.c

```diff
@@ -896,6 +896,11 @@ crt_init_opt(crt_group_id_t grpid, uint32_t flags, crt_init_options_t *opt)
 	D_FREE(domain0);
 	D_FREE(provider_str0);
 	D_FREE(auth_key0);
+	d_freeenv_str(&port_str);
```

Comment:
why not just save the return from d_agetenv and call these later rather than adding the extra dup?

Reply:
Tried to find the less disruptive solution without introducing new variables, and I also not think of this approach.
At the end, I really prefer your solution which prevent useless extra memory allocation. Thus, I will update all the patch with your proposal.
- [ ] Remove useless `strdup()` with saving `d_agetenv()` output result

Reply:
> Tried to find the less disruptive solution without introducing new variables, and I also did not think of this approach. At the end, I really prefer your solution which prevents useless extra memory allocation. Thus, I will update all the patch with your proposal.
> 
> * [x]  Remove useless `strdup()` with saving `d_agetenv()` output result

Fixed with commit 761ecbe70824457f694903256bc7465c20e85721